testing/test_warnings.py

```diff
@@ -521,9 +521,16 @@ def test_invalid_regex_in_filterwarning(self, pytester: Pytester) -> None:
             """
         )
         result = pytester.runpytest_subprocess()
+        assert result.ret == pytest.ExitCode.USAGE_ERROR
         result.stderr.fnmatch_lines(
             [
-                "*Invalid regex '*': nothing to repeat at position 0*",
+                "ERROR: while parsing the following warning configuration:",
+                "",
+                "  ignore::DeprecationWarning:*",
+                "",
+                "This error occurred:",
+                "",
+                "Invalid regex '*': nothing to repeat at position 0",
```

Comment:
I think you need to escape this star or it will treat it as a wildcard/fnmatch syntax. I'm not sure how to escape it though


```suggestion
                r"Invalid regex '\*': nothing to repeat at position 0",
```

Reply:
makes sense, I will check what I can do here.

Reply:
we can escape * using  [*]
ref: https://docs.python.org/3/library/fnmatch.html